pretrain/openwebtext.py

```diff
@@ -158,30 +158,30 @@ def train(fabric: L.Fabric, state: dict, train_dataloader: DataLoader, val_datal
             optimizer.zero_grad()
             state["step_count"] += 1
 
-        total_lengths += input_ids.size(1)
-        if state["iter_num"] % log_interval == 0:
+        if iter_num % log_interval == 0:
             loss_item = loss.item()  # expensive device-to-host synchronization
             t1 = time.perf_counter()
             throughput.update(
                 time=t1 - total_t0,
-                samples=(state["iter_num"] + 1) * micro_batch_size,
-                lengths=total_lengths,
+                batches=iter_num,
+                samples=iter_num * micro_batch_size,
+                lengths=iter_num * micro_batch_size * model.max_seq_length,
```

Comment:
is `model.max_seq_length == input_ids.size(1)` in this case? It needs to be block size of the data.

Reply:
Let's use `model.config.block_size` here?

Reply:
> is model.max_seq_length == input_ids.size(1) in this case? It needs to be block size of the data.

Yes because the current pertaining setup takes batches of T=block size=max_seq_length

> Let's use model.config.block_size here?

They are the same in this script

Reply:
That block size and max_seq_length is just not conceptually the same. Here we are pretraining, so we choose the block size as the primary variable because that's how long the sequences are we load from the dataloader. So here, it is tied to the value in the dataloader, not the model. And so even if the values  end up the same, I would choose block size

Reply:
In lit-gpt, the `block_size` is the reference value tied to the config, but the `max_seq_length` is what is used at runtime inside the model and in the scripts.

During pretraining, the dataloader is configured to return sequences of `block_size` length, but this is not required during fine-tuning. From this perspective, using `max_seq_length` is more reliable if somebody was to change the dataloading setup in the script.

In any case, this is bikeshedding. It doesn't matter functionally here